src/server/listener.cc

```diff
@@ -170,6 +170,10 @@ void Listener::bind(const Address &address) {
     throw std::runtime_error(strerror(errno));
   }
 
+  if (fd == -1) {
```

Comment:
This should not be needed, but I do see a bug above.  On line #158, after "close(fd)", fd should be set back to -1.  Because this might be on the last iteration through the for loop, so the loop would exit with fd>=0, but unbound.  Your code check here will not catch that.  When the for loop exits, either "addr" is not null (and fd is bound, listening and >=0), or addr is null and fd should be ignored.

Reply:
@dennisjenkins75 You are totally right, my check is useless in this case. On the other hand I don't see any problems on line 158. If it is the last iteration then `addr` will be `nullptr` and the exception will be thrown and it doesn't matter what's the value local variable `fd` has.